### PR TITLE
Fix typo in application of y value BCs

### PR DESCRIPTION
**Pull request by ali-ramadhan**

### CODE REVIEW

**Comment by glwagner** [comment]
Hmm... I don’t think this is a typo. The functions *_gradient for “west”, “south”, and “bottom” are *identical*. Therefore we should use the alias “left” for all three to avoid copy-pasting identical functions. The same applies to the three “right” boundaries: “east”, “north”, and “top”.

If we want to use matching names in other functions then we should use ‘const’ aliases. I think copy-pasting function definitions is bad practice that decreases maintainability and can make bugs harder to catch and fix.

**Comment by glwagner** [comment]
This is not part of this PR, but these 4 functions have identical definitions:

```julia
@inline bottom_gradient(bc::GBC, c¹, Δ, i, j, args...) = getbc(bc, i, j, args...)
@inline    top_gradient(bc::GBC, cᴺ, Δ, i, j, args...) = getbc(bc, i, j, args...)
@inline south_gradient(bc::GBC, c¹, Δ, i, k, args...) = getbc(bc, i, k, args...)
@inline north_gradient(bc::GBC, cᴺ, Δ, i, k, args...) = getbc(bc, i, k, args...)
```

File: src/BoundaryConditions/apply_value_gradient_bcs.jl [hunk 1]
```diff
@@ -36,8 +36,8 @@ end
 @inline bottom_gradient(bc::VBC, c¹, Δ, i, j, args...) = ( c¹ - getbc(bc, i, j, args...) ) / (Δ/2)
 @inline    top_gradient(bc::VBC, cᴺ, Δ, i, j, args...) = ( getbc(bc, i, j, args...) - cᴺ ) / (Δ/2)
 
-@inline  left_gradient(bc::VBC, c¹, Δ, i, k, args...) = ( c¹ - getbc(bc, i, k, args...) ) / (Δ/2)
-@inline right_gradient(bc::VBC, cᴺ, Δ, i, k, args...) = ( getbc(bc, i, k, args...) - cᴺ ) / (Δ/2)
+@inline south_gradient(bc::VBC, c¹, Δ, i, k, args...) = ( c¹ - getbc(bc, i, k, args...) ) / (Δ/2)
```
Comment (glwagner) [inline src/BoundaryConditions/apply_value_gradient_bcs.jl]: This function's definition is identical to `bottom_gradient`.

I think we should have two functions, `left_gradient` and `right_gradient`. `left_gradient` should be called in `_fill_bottom_halo!(c, bc::Union{VBC, GBC}, grid, args...)`, `_fill_south_halo!(c, bc::Union{VBC, GBC}, grid, args...)`, and `_fill_west_halo!(c, bc::Union{VBC, GBC}, grid, args...)`, while `right_gradient` should be called in the same functions for `top`, `north`, and `east`.

and 

```

**Comment by ali-ramadhan** [comment]
Good observation. I've simplified the code down to just `left_gradient` and `right_gradient` so there's still one duplicate definition but it'll be much better than having 5 duplicates once we support `Value` and `Gradient` boundary conditions in x.

**Pull request merged**